core/src/main/scala/kafka/log/Log.scala

```diff
@@ -1050,7 +1050,8 @@ class Log(@volatile private var _dir: File,
                      leaderEpoch: Int,
                      origin: AppendOrigin = AppendOrigin.Client,
```

Comment:
It would be helpful to have a test case which verifies `appendAsLeader` with the new append origin.

Reply:
fixed